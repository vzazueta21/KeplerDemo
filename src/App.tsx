import React from 'react';
import {StyleSheet, Text, View} from 'react-native';

// prettier-ignore
export const App = () => {

  return (
    <View style={styles.container}>
      <View
        style={styles.header}
        testID="sampleHeader">
        <Text style={styles.headerLabel}>Welcome to Accedo's Kepler Demo</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#283593',
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    alignItems: 'center',
    backgroundColor: '#000000',
    borderColor: 'navy',
    borderRadius: 10,
    borderWidth: 1,
    paddingVertical: 12,
    paddingHorizontal: 32,
  },
  headerLabel: {
    color: 'white',
    fontSize: 22,
    fontFamily: 'Amazon Ember',
  },
  colorLabel: {
    color: 'white',
    position: 'absolute',
    top: 32,
    right: 32,
    fontSize: 32,
  },
});
