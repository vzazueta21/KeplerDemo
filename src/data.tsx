const DATA = [
  {
    id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
    name: 'First Item',
  },
  {
    id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
    name: 'Second Item',
  },
  {
    id: '58694a0f-3da1-471f-bd96-145571e29d72',
    name: 'Third Item',
  },
  {
    id: '58694a0f-3da1-471f-bd96-143208390ufiud',
    name: 'Fourth Item',
  },
  {
    id: '58694a0f-3da1-471f-bd96-1418eruhuiwhf0w',
    name: 'Fifth Item',
  },
  {
    id: '58694a0f-3da1-471f-bd96-284852eed0',
    name: 'Sixth Item',
  },
];

export default DATA;
