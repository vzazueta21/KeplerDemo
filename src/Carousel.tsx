import React from 'react';
import {FlatList, StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import {DATA} from './data';

const Carousel = ({title, onPressItem}) => {
  const renderItem = ({item}) => {
    return (
      <TouchableOpacity onPress={onPressItem} style={styles.item}>
        <Text style={styles.itemName}>{item.name}</Text>
      </TouchableOpacity>
    );
  };

  return (
    <View style={styles.carouselContainer}>
      <Text style={styles.carouselTitle}>{title}</Text>
      <FlatList
        horizontal={true}
        data={DATA}
        renderItem={renderItem}
        keyExtractor={item => item.id}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  carouselContainer: {
    marginBottom: 30,
    justifyContent: 'flex-start',
  },
  carouselTitle: {
    fontSize: 40,
    color: '#FFFFFF',
  },
  item: {
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
    backgroundColor: '#6e3b6e',
    height: 200,
    width: 350,
  },
  itemName: {
    fontSize: 32,
    color: 'white',
  },
});

export default Carousel;
