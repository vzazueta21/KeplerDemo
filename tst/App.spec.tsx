/**
 * Copyright (c) 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */
import 'react-native';

import {fireEvent, render} from '@testing-library/react-native';
import * as React from 'react';

import {ComAmazonKepleraccedoMain} from '../src/App';

describe('Template App Snapshot tests', () => {
  it('Initial App screen', () => {
    // prettier-ignore
    const screen = render(<ComAmazonKepleraccedoMain />);
    expect(screen).toMatchSnapshot();
  });
  it('App screen after button press', () => {
    // prettier-ignore
    const screen = render(<ComAmazonKepleraccedoMain />);
    const button = screen.getByTestId('sampleButton');
    fireEvent.press(button);
    expect(screen).toMatchSnapshot();
  });
});
