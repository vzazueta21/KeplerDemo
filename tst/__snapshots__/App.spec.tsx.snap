// Jest Snapshot v1, https://goo.gl/fbAQLP
/*
 * Copyright (c) 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */

exports[`Template App Snapshot tests App screen after button press 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1A237E",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#303030",
        "borderColor": "navy",
        "borderRadius": 10,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 12,
      }
    }
    testID="sampleButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Amazon Ember",
          "fontSize": 22,
        }
      }
    >
      Change Color
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 32,
        "position": "absolute",
        "right": 32,
        "top": 32,
      }
    }
  >
    #1A237E
  </Text>
</View>
`;

exports[`Template App Snapshot tests Initial App screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#283593",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#303030",
        "borderColor": "navy",
        "borderRadius": 10,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 12,
      }
    }
    testID="sampleButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Amazon Ember",
          "fontSize": 22,
        }
      }
    >
      Change Color
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 32,
        "position": "absolute",
        "right": 32,
        "top": 32,
      }
    }
  >
    #283593
  </Text>
</View>
`;
